### ISSUE TITLE
Provided example does not work because the default working directory has unexpected value.

### ISSUE THREAD

**Issue by shurcooL** [shurcooL/trayhost#9]
On OS X, when you run an app bundle, the working directory of the executed process is the root directory (`/`), and not the app bundle's `Example.app/Contents/Resources` folder.

As a result, the provided example is unlikely to work, because the library needs a valid image file as the [2nd parameter to `trayhost.Initialize`](http://godoc.org/github.com/shurcooL/trayhost#Initialize), and if the user places an image like `icon@2x.png` in `Resources` folder and writes code:

``` Go
// Tray icon.
iconData, err := ioutil.ReadFile("icon@2x.png")
if err != nil {
    log.Fatalln(err)
}
```

They'll get an error like:

```
2016/08/01 20:51:47 open icon@2x.png: no such file or directory
```

Because `/icon@2x.png` file doesn't exist.

/cc @jacobrosenthal

**Comment by shurcooL** [comment]
I see two possible ways to resolve this.

One is to document this quirk about OS X app bundles and require the user to deal with it. Perhaps provide some helper func to fetch the "app bundle" or equivalent directory.

Another way is to make `trayhost.Initialize` do the following:

```
// Initialize sets up the application properties.
//
// On OS X, it changes the current directory to the app bundle's Contents/Resources
// directory, if present.
```

I like the 2nd approach more because it's a more friendly to use API.

It's also the API that GLFW C library uses (unless one disables that with a compile-time option): /cc @elmindreda

http://www.glfw.org/docs/latest/group__init.html#ga317aac130a235ab08c6db0834907d85e

> Remarks
> - **OS X:** This function will change the current directory of the application to the `Contents/Resources` subdirectory of the application's bundle, if present. This can be disabled with a compile-time option.

Any strong objections or comments on that? /cc @jacobrosenthal @bradfitz

**Comment by shurcooL** [comment]
Well, I just realized that's not going to work, because the `trayhost.Initialize` func requires you to provide it with the image data, so you'd have to try to read the file **before** calling `trayhost.Initialize`. So even if changes working directory, it'd be too late.

```Go
// Tray icon.
iconData, err := ioutil.ReadFile("app-icon@2x.png")
if err != nil {
	log.Fatalln(err)
}

trayhost.Initialize("Example App", iconData, menuItems) // Too late to change directory!
```

**Comment by shurcooL** [comment]
In https://github.com/shurcooL/trayhost/issues/7#issuecomment-236787476, I mentioned another solution, which I actually currently use, of just using a bash script to set the working directory and then launch the binary.

In https://github.com/shurcooL/trayhost/pull/8#issuecomment-237263674, @jacobrosenthal wrote:

> Im not sure how hacky that is long run, but it does work.

(Which I confirmed [here](https://github.com/jacobrosenthal/trayhost/commit/6e1c310978cc5b03ace193fe3e7d9df76359aeb2#commitcomment-18514976).)

I agree that it works, that's how I've been using Instant Share.app for years. It is very practical, because it lets you add/change flags to the Go binary very easily, as well as redirect its stdout/stderr to a log file, etc. Also, bash is guaranteed to be present on OS X.

So maybe I should just use it as a solution... While it feels gross to rely on a bash script, I can't seem to think of a much better way (especially if you want to set flags or redirect stdout/stderr).

**Comment by shurcooL** [comment]
An idea to resolve this more nicely than a bash shell script is use a new Go 1.8 addition:

https://beta.golang.org/pkg/os/#Executable

I could recommend an `init()` on macOS that looks like this:

```Go
func init() {
	ep, err := os.Executable()
	if err != nil {
		log.Fatalln("os.Executable:", err)
	}
	err = os.Chdir(filepath.Join(filepath.Dir(ep), "..", "Resources"))
	if err != nil {
		log.Fatalln("os.Chdir:", err)
	}
}
```

**Comment by shurcooL** [comment]
I think this specific issue has been resolved by 9bded1aa5ec00ffea58a010967da7d05eb5135cd. Closing.
